contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,847 @@
+# StatefulSet Updates
+
+**Author**: kow3ns@
+
+**Status**: Proposal
+
+## Abstract
+Currently (as of Kubernetes 1.6), `.Spec.Replicas`, and 
+`.Spec.Template.Containers` are the only mutable fields of the 
+StatefulSet API object. Updating `.Spec.Replicas` will scale the number of Pods 
+in the StatefulSet. Updating `.Spec.Template.Containers` causes all subsequently 
+created Pods to have the specified containers. In order to cause the 
+StatefulSet controller to apply its updated `.Spec`, users must manually delete 
+each Pod. This manual method of applying updates is error prone. The 
+implementation of this proposal will add the capability to perform ordered,
+automated, sequential updates. 
+
+## Affected Components
+1. API Server
+1. Kubectl
+1. StatefulSet Controller
+1. StatefulSetSpec API object
+1. StatefulSetStatus API object
+
+## Use Cases
+Upon implementation, this design will support the following in scope use cases, 
+and it will not rule out the future implementation of the out of scope use 
+cases.
+
+### In Scope
+- As the administrator of a stateful application, in order to vertically scale 
+my application, I want to update resource limits or requested resources.
+- As the administrator of a stateful application, in order to deploy critical 
+security updates, break fix patches, and feature releases, I want to update 
+container images.
+- As the administrator of a stateful application, in order to update my 
+application's configuration, I want to update environment variables, container 
+entry point commands or parameters, or configuration files.
+- As the administrator of the logging and monitoring infrastructure for my 
+organization, in order to add logging and monitoring side cars, I want to patch
+containers to add images.
+
+### Out of Scope
+- As the administrator of a stateful application, in order to increase the 
+applications storage capacity, I want to update PersistentVolumes.
```

Comment:
The idea would be to allow the VolumeClaimsTemplate to be updated to allow for resizing the PVs attached to the PVC. For EBS and PD, users can manually resize PVs after creation and binding, but with no way to mutate the PVCs generate from the VolumeClaimsTemplate subsequent PVCs will created with the prior capacity. At some point, it may be useful to be able to automatically trigger the resize where the provisioner supports such a feature.